### DISCUSSION TITLE
Arbitrary cargo build arguments

### DISCUSSION THREAD

**Discussion by mightyiam** [90-008/nix-cargo-integration#157] [Q&A] [answered]
Is there a way to pass arbitrary cargo build arguments, such as `--deny-warnings`, please?

**Comment by 90-008** [comment]
You can use testFlags or buildFlags to pass arbitrary arguments to cargo, just use those under mkDerivation in drvConfig and depsDrvConfig

**Reply by mightyiam** [reply]
Thank you. `testFlags` doesn't seem to exist. `buildFlags` doesn't seem to apply.

Where are these defined, please?

**Reply by 90-008** [reply, marked as answer]
Sorry, I mixed up some of the options, you have to define `rust-crane.testFlags` and `rust-crane.buildFlags` under `drvConfig` and `depsDrvConfig`. The `drvConfig` are passed to dream2nix as is basically, and in dream2nix it's configured through `rust-crane`.

**Reply by mightyiam** [reply]
And I apologize. There's no such cargo argument `--deny-warnings`. There's `cargo clippy -- --deny warnings`. And there's `RUSTFLAGS="--deny warnings"` which seems to affect most cargo commands, but not `cargo doc`. For that, one needs `RUSTDOCFLAGS` with the same value. So, I will mark this an answered.
